test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,568 @@
+            cmd ="get-foo22h";
+            expected = $null
+        },
+        @{
+            name = "Invoking with script block";
+            cmd = ". { param([Parameter(position=2)] `$a, [Parameter(position=1)]`$b); `$a; `$b} a b";
+            expected = @('b','a')
+        }        
+    )
+
+    It '<name>' -TestCases:$testData{
+        param($name, $cmd, $expected)
+        
+        $actual = Invoke-Expression $cmd
+        if(($expected -ne $null ) -and ($expected.GetType().IsArray))
+        {
+            $actual.Count | Should Be $expected.Count
+            for($i = 0; $i -lt $expected.Count; $i++)
+            {
+                $actual[$i] | Should Be $expected[$i]
+            }
+        }
+        else
+        {
+            $actual | Should Be $expected
+        }
+    }
+    
+    $testData1 = @(
+        @{
+            name = 'default values not converted properly';    
+            cmd = 'get-foo22a';    
+            expected='System.Reflection.MemberTypes'
+        },
+        @{
+            name = 'default values not converted properly';    
+            cmd = 'get-foo22b';    
+            expected='System.Reflection.MemberTypes'
+        }
+        )
+
+    It 'Validate Type: <name>' -TestCases:$testData1 {
+        param($name, $cmd, $expected)            
+                        
+        $actual = Invoke-Expression $cmd        
+        $actual | Should BeOfType $expected 
+    }
+
+    $testData2 = @(
+        @{
+            name = 'Parameter Set cannot be resolved';
+            cmd = 'get-foo8 -a a -b b c d';   
+            ErrorId = 'AmbiguousParameterSet,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+         },
+        @{
+            name = 'Null Not Allowed';
+            cmd = "get-foo13 -a `$null";   
+            ErrorId = 'ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+         },
+        @{
+            name = 'Empty String Not Allowed';
+            cmd = "get-foo15 -a ''";   
+            ErrorId = 'ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+         },
+        @{
+            name = 'Empty collection not allowed';
+            cmd = "get-foo17 -a `@()";
+            ErrorId = 'ParameterArgumentValidationErrorEmptyArrayNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+         },
+        @{
+            name = "Binding array of string to array of bool fails "
+            cmd ="get-foo21 'a','b'";
+            ErrorId = 'ParameterArgumentTransformationError,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+        }
+     )
+
+    It '<name>' -TestCases $testData2 {
+            param($name, $cmd, $ErrorID)
+        try {
+            Invoke-Expression $cmd
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be $ErrorId
+        }
+    }   
+    
+    #know issue
```

Comment:
fixed.

---
In reply to: [78272031](https://github.com/Powershell/Powershell/pull/2218#discussion_r78272031) [](ancestors = 78272031)